walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,113 +15,81 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
 import java.util.Arrays;
 import java.util.ArrayList;
 import java.util.HashSet;
 import java.util.Set;
 
-public final class FindMeetingQuery {        
+public final class FindMeetingQuery {   
 
-    public Collection <TimeRange> query(Collection <Event> events, MeetingRequest request) {
-      long calculation = 0;
-      long duration = request.getDuration();
-      Collection<String> mandatoryAttendees = request.getAttendees();
+    private static int MAX_DURATION = 60 * 24;
+
+    // Returns sorted TimeRanges that have attendees in the attendees set.
+    private static ArrayList<TimeRange> getBusyTimeRangesForAttendees(Collection<Event> events, Collection<String> mandatoryAttendees) {   
+      Set<String> attendeesSet = new HashSet<String>();
+      for(String attendee : mandatoryAttendees) {
+        attendeesSet.add(attendee);
+      }
+
+      ArrayList<TimeRange> timeRangesWithAttendees = new ArrayList <TimeRange> ();
+      for(Event event: events) {
+        Collection<String> eventAttendees = event.getAttendees();
+        TimeRange when = event.getWhen();
+        for(String attendee : eventAttendees) {   
+          if(attendeesSet.contains(attendee)) {
+            timeRangesWithAttendees.add(when);
+            break;
+          } 
+        }
+      }
+      Collections.sort(timeRangesWithAttendees, TimeRange.ORDER_BY_START);
+      return timeRangesWithAttendees;
+    }
+
+    //Calculates the availbleTimes from the series of events
+    private static ArrayList <TimeRange> findAvailableTimes(Collection<Event> events, Collection<String> mandatoryAttendees, long duration) {
```

Comment:
nit: This line is long; Consider using a line-guide (vscode lets you configure one) to restrict your lines to 80 characters (that's what we use internally). I can be annoying and look uggly at first, but it's a lot easier to read if you don't have to scan from side-to-side as much.